libs/sdk-ui-charts/src/highcharts/adapter/plugins/adjustTickAmount.ts

```diff
@@ -5,17 +5,23 @@
  *      https://github.com/highcharts/highcharts/blob/b54fe33d91c0d1fd7da009aaa84af694f15cffad/js/parts/Axis.js#L4214
  *
  * Modified by binh.nguyen@gooddata.com to support zero alignment
+ *
+ * Relying on undocumented internal properties of Highcharts.Axis
+ *  - visible
+ *  - tickAmount
+ *  - tickInterval
+ *  - transA
+ *  - opposite
  */
 
 import isNil from "lodash/isNil";
-import get from "lodash/get";
 import Highcharts from "../../lib";
-import { IHighchartsAxisExtend } from "../../typings/extend";
 import { isLineChart } from "../../chartTypes/_util/common";
+import { UnsafeInternals } from "../../typings/unsafe";
 
 interface IBaseAndAlignedAxes {
-    baseYAxis: IHighchartsAxisExtend;
-    alignedYAxis: IHighchartsAxisExtend;
+    baseYAxis: Highcharts.Axis;
```

Comment:
Using Highcharts.Axis & UnsafeInternals instead of the dedicated type with all the internals included